packages/client/src/components/GameBoard/GameBoard.tsx

```diff
@@ -85,13 +89,38 @@ const GameBoard: React.FC = () => {
     }
   }
 
+  const [isFullscreenEnabled, setFullscreenEnabled] = useState<boolean>(false)
   return (
-    <>
+    <FullscreenProvider
+      enabled={isFullscreenEnabled}
+      onChange={(isFullscreenEnabled: boolean) =>
+        setFullscreenEnabled(isFullscreenEnabled)
+      }>
       <div style={styles}>
+        <Button
+          className="fullscreen-btn"
+          aria-label="OpenInFull"
+          onClick={() => setFullscreenEnabled(!isFullscreenEnabled)}
```

Comment:
[Исправление]
Для изменения состояния правильнее использовать параметр сета `setFullscreenEnabled`, а не само текущее значение `isFullscreenEnabled`. [Вот тут](https://stackoverflow.com/questions/59073180/react-usestate-using-previous-state-vs-not-using-previous-state) на стековерфлоу есть примеры, что именно будет не так.
```suggestion
          onClick={() => setFullscreenEnabled(prev => !prev)}
```